sql-native/src/main/java/org/ballerinalang/sql/nativeimpl/CallProcessor.java

```diff
@@ -0,0 +1,436 @@
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.sql.nativeimpl;
+
+import io.ballerina.runtime.api.TypeTags;
+import io.ballerina.runtime.api.creators.TypeCreator;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.types.StructureType;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BStream;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BTypedesc;
+import io.ballerina.runtime.transactions.TransactionResourceManager;
+import org.ballerinalang.sql.Constants;
+import org.ballerinalang.sql.datasource.SQLDatasource;
+import org.ballerinalang.sql.exception.ApplicationError;
+import org.ballerinalang.sql.parameterprocessor.ResultParameterProcessor;
+import org.ballerinalang.sql.parameterprocessor.StatementParameterProcessor;
+import org.ballerinalang.sql.utils.ColumnDefinition;
+import org.ballerinalang.sql.utils.ErrorGenerator;
+import org.ballerinalang.sql.utils.ModuleUtils;
+
+import java.io.IOException;
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+
+import static org.ballerinalang.sql.Constants.CONNECTION_NATIVE_DATA_FIELD;
+import static org.ballerinalang.sql.Constants.DATABASE_CLIENT;
+import static org.ballerinalang.sql.Constants.PROCEDURE_CALL_RESULT;
+import static org.ballerinalang.sql.Constants.QUERY_RESULT_FIELD;
+import static org.ballerinalang.sql.Constants.RESULT_SET_COUNT_NATIVE_DATA_FIELD;
+import static org.ballerinalang.sql.Constants.RESULT_SET_TOTAL_NATIVE_DATA_FIELD;
+import static org.ballerinalang.sql.Constants.STATEMENT_NATIVE_DATA_FIELD;
+import static org.ballerinalang.sql.Constants.TYPE_DESCRIPTIONS_NATIVE_DATA_FIELD;
+import static org.ballerinalang.sql.utils.Utils.getColumnDefinitions;
+import static org.ballerinalang.sql.utils.Utils.getDefaultRecordType;
+import static org.ballerinalang.sql.utils.Utils.getSqlQuery;
+import static org.ballerinalang.sql.utils.Utils.updateProcedureCallExecutionResult;
+
+/**
+ * This class holds the utility methods involved with executing the call statements.
```

Comment:
```suggestion
 * This class holds the utility methods involved with executing the call statements.
 *
 * @since 0.5.6
```

Reply:
I will change it

Reply:
I added it. Commit id is f329209456c93f66091ef0d1ec324a9ad137aaec